### PR TITLE
ebnf select

### LINKED ISSUE #17274: sql-statements/sql-statement-select.md

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #17274

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-select.md [hunk 1]
```diff
@@ -10,82 +10,51 @@ The `SELECT` statement is used to read data from TiDB.
...
+SelectStmtBasic ::=
+    "SELECT" SelectStmtOpts Field ("," Field)* ( "HAVING" Expression)?
 
-**SelectStmtFieldList:**
+SelectStmtFromDualTable ::=
+    "SELECT" SelectStmtOpts Field ("," Field)* "FROM" "DUAL" WhereClauseOptional
 
-![SelectStmtFieldList](/media/sqlgram/SelectStmtFieldList.png)
+SelectStmtFromTable ::=
+    "SELECT" SelectStmtOpts Field ("," Field)* "FROM" TableRefsClause WhereClauseOptional SelectStmtGroup HavingClause WindowClauseOptional
```
Comment (kennytm) [inline sql-statements/sql-statement-select.md:25]: AFAIK `("HAVING" Expression)?` and `HavingClause` are equivalent. The two references from `SelectStmtBasic` and `SelectStmtFromTable` should be using the same pattern.

Also I suggest *not* hiding the lone `?` behind `HavingClause` and `WhereClauseOptional` and `SelectStmtGroup` and `WindowClauseOptional` etc.

File: sql-statements/sql-statement-select.md [hunk 2]
```diff
@@ -10,82 +10,51 @@ The `SELECT` statement is used to read data from TiDB.
...
+WindowClauseOptional ::=
+    "WINDOW" WindowDefinition ("," WindowDefinition)*
 
-![WindowClauseOptional](/media/sqlgram/WindowClauseOptional.png)
-
-**TableSampleOpt**
-
-```ebnf+diagram
 TableSampleOpt ::=
     'TABLESAMPLE' 'REGIONS()'
```
Comment (kennytm) [inline sql-statements/sql-statement-select.md:59]:
```suggestion
    'TABLESAMPLE' 'REGIONS' '(' ')'
```

File: sql-statements/sql-statement-select.md [hunk 3]
```diff
@@ -10,82 +10,51 @@ The `SELECT` statement is used to read data from TiDB.
...
+    WhereClause?
 
-![OrderByOptional](/media/sqlgram/OrderByOptional.png)
+SelectStmtGroup ::=
+    GroupByClause?
 
-**SelectStmtLimit:**
+SelectStmtLimit ::=
+    ("LIMIT" LimitOption ( ("," | "OFFSET") LimitOption )?
+| "FETCH" ("FIRST" | "NEXT") LimitOption ("ROW" | "ROWS") "ONLY" )
```
Comment (kennytm) [inline sql-statements/sql-statement-select.md:46]:
```suggestion
| "FETCH" ("FIRST" | "NEXT") LimitOption? ("ROW" | "ROWS") "ONLY" )
```

**Review by Oreoxmt — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**